src/run_autodiscovery.py

```diff
@@ -0,0 +1,237 @@
+        object.__setattr__(self, "description", kwargs.get("description", ""))
+        object.__setattr__(self, "unit", kwargs.get("unit", "Unspecified"))
+        object.__setattr__(self, "tags", kwargs.get("tags", []))
+        object.__setattr__(self, "dimensions", kwargs.get("dimensions", []))
+        object.__setattr__(self, "metricProperties", kwargs.get("metricProperties", []))
+
+        object.__setattr__(self, "schemaId", kwargs.get("schemaId","builtin:metric.metadata"))
+        object.__setattr__(self,"scope","metric-"+kwargs.get("key", "").replace(":", "."))
+
+    def to_json(self):
+        return json.dumps({
+            "scope": self.scope,
+            "schema.Id": self.schema_id,
+            "value": {
+                "displayName": self.display_name,
+                "description": self.description,
+                "unit": self.unit,
+                "tags": self.tags,
+                "metricProperties": self.metric_properties,
+                "dimensions": self.dimensions
+            }
+        })
+
+
+def unit_caster(unit: str) -> str:
+    if unit in ["1","count","Count", "{operation}","{packet}" ,"{packets}" ,"{request}" ,"{port}" ,"{connection}", "{devices}" ,"{errors}" ,"{cpu}", "{query}", "{inode}"]:
+        return "Count"
+    elif unit == "s" or unit =="s{idle}" or unit == "sec":
+        return "Second"
+    elif unit == "By" or unit == "Byte" or unit == "bytes":
+        return "Byte"
+    elif unit == "By/s":
+        return "BytePerSecond"
+    elif unit == "kBy":
+        return "KiloBytePerSecond"
+    elif unit == "GBy.s":
+        return "GigaBytePerSecond"
+    elif unit == "GiBy" or unit == "GBy":
+        return "GigaByte"
+    elif unit == "MiBy":
+        return "MegaByte"
+    elif unit == "ns":
+        return "NanoSecond"
+    elif unit == "us" or unit == "usec":
+        return "MicroSecond"
+    elif unit == "ms" or unit == "milliseconds":
+        return "MilliSecond"
+    elif unit == "us{CPU}":
+        return "MicroSecond"
+    elif unit == "10^2.%":
+        return "Percent"
+    elif unit == "%":
+        return "Percent"
+    elif unit == "percent":
+        return "Percent"
+    elif unit == "1/s":
+        return "PerSecond"
+    elif unit == "frames/seconds":
+        return "PerSecond"
+    elif unit == "s{CPU}":
+        return "Second"
+    elif unit == "s{uptime}":
+        return "Second"
+    elif unit == "{dBm}":
+        return "DecibelMilliWatt"
+    else:
+        return "Unspecified"
+    
+
+def type_metrickind_caster(metrickind: str, valuetype: str) -> str:
+    if metrickind == "GAUGE":
+        return "gauge"
+    if metrickind == "DELTA" and valuetype == "DISTRIBUTION":
+        return "gauge"
+    if metrickind == "DELTA" and valuetype != "DISTRIBUTION":
+        return "count,delta"
+    else:
+        return ""
+
+    
+def metric_name_parse(metric_name: str,metric_type: str) -> str:
```

Comment:
parse = read 
you are kinda transforming, not really reading? Please make it more descriptive. I think it is "change_metric_key_from_gcp_format_to_dt_format" (I am not sure though if this is the functionality)